update version and dependencies

jgulotta opened the pull request:
This is meant to accompany https://github.com/slack-rs/slack-rs-api/pull/26

Like the other PR, tests pass and I was able to use it in my app successfully.

BenTheElder commented:
Restarting the Travis build after merging https://github.com/slack-rs/slack-rs-api/pull/26.

BenTheElder commented:
I think we'll have to cut a release for slack-rs-api before the tests will pass, I will take a look at this later.

jgulotta commented:
Oh it's probably because there's already a `slack_api = 0.14.0` on crates.io. I didn't notice before. Yeah I think `slack_api` will need to be `0.14.1` or `0.15.0`. I can update the PR whenever you decide which version number you'd like.

BenTheElder commented:
I think in general we have a problem with different versions of hyper in
both, we probably just need to use hyper exported from the Api crate in the
slack crate to avoid hyper updates being a breaking change in the future.
We'll need to look at that first.
On May 18, 2016 14:46, "Jacob Gulotta" <notifications@github.com> wrote:

> Oh it's probably because there's already a slack_api = 0.14.0 on crates.io.
> I didn't notice before. Yeah I think slack_api will need to be 0.14.1 or
> 0.15.0. I can update the PR whenever you decide which version number
> you'd like.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/slack-rs/slack-rs/pull/55#issuecomment-220121363>
>

jgulotta commented:
Sent https://github.com/slack-rs/slack-rs-api/pull/27 and updated this PR to used the re-exported hyper

mthjones commented:
Thanks for the PRs! But I'm not sure I follow here. The issue is that the `slack_api` version you bumped the dependency to still relies on `hyper 0.8.0`, which is obviously going to break when this tries to use `hyper 0.9.4`, since they're different packages. The solutions to that are:

* stick to `hyper 0.8.0`. What new features or bugfixes does `hyper 0.9.4` give us?
* wait for the updated `slack_api` to release that uses `hyper 0.9.4`, then update this to use that new version
* wait for [this PR](https://github.com/slack-rs/slack-rs-api/pull/24) to land and release, then implement a new `SlackWebRequestSender` that wraps whatever hyper version you want to use

I'm not sure I agree that re-exporting hyper is the appropriate solution, especially after the aforementioned PR lands.

mthjones commented on src/lib.rs:
```diff
@@ -17,16 +17,16 @@
 extern crate websocket;
 extern crate rustc_serialize;
-extern crate slack_api;
+pub extern crate slack_api;
```
Why does this need to be re-exported? We already export it as `api` below.

jgulotta replied:
Without it you get an E0446 warning

mthjones replied:
Could you paste (or make a gist of) where it said the error was? I see the error references private types in public interfaces, but I just don't see where that's an issue here.

jgulotta replied:
Sorry, that was not well-worded. E0446 is the ultimate reason for the warning. Here's the text on rustc 1.8.0:

```
src/lib.rs:27:9:` 27:25 warning: extern crate `slack_api` is private, and cannot be reexported (error E0364), consider declaring with `pub`, #[warn(private_in_public)] on by default
src/lib.rs:27 pub use slack_api as api;
                      ^~~~~~~~~~~~~~~~
src/lib.rs:27:9: 27:25 warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
src/lib.rs:27:9: 27:25 note: for more information, see the explanation for E0446 (`--explain E0446`)
```

BenTheElder commented:
Waiting for that PR makes more sense to me, I forgot about that. Haven't been using slack and I got pretty wrapped up with switching apartments and classes myself.
I'll do my best to catch up on both crates tonight.

jgulotta commented:
>  the slack_api version you bumped the dependency to still relies on hyper 0.8.0

I sent the PR to `slack_api` to up the version of hyper without realizing 0.14.0 was already published to crates.io. This PR mentions that both the `slack_api` and `slack_rs` versions need to be updated.

> I'm not sure I agree that re-exporting hyper is the appropriate solution, especially after the aforementioned PR lands.

Waiting for the api PR#24 makes sense to me too if it's actually landing soon. It looked like it had been stalled for about a month. I'm not sure these PRs preclude that one from actually being completed, and in general quick fixes seem okay to me when you have the minimal stability guarantees of a 0.x version.

In the end the only true motivation for these is that I hate seeing results when I run `cargo outdated` and my only outdated ones were coming from slack. I fixed them for myself locally and thought I'd see if you guys wanted them too. If you'd rather wait for the refactor and not do these that's totally fine by me. Feel free to close 'em!

mthjones commented:
`slack_api@0.15.0` has been released with the new hyper version and also the generic client changes.

jgulotta commented:
Great. I'll check that out. Maybe I replace this PR with one which uses the new API.

jgulotta commented:
Updated to slack_api 0.15.0. I don't think it makes sense to depend on the version of an optional transitive dependency so I removed that. This now is just version updates and silencing the compiler warning.

mthjones commented:
With that compiler warning, instead of exporting the `slack_api` crate two times (once as `api`, once as `slack_api`), perhaps the latter `pub use` could be removed and the `extern crate` updated to `extern crate slack_api as api`?

jgulotta commented:
Makes sense to me. Done.

BenTheElder commented:
LGTM, this should fix https://github.com/slack-rs/slack-rs/issues/57.
